docker_compose_swarm_mode.py

```diff
@@ -537,6 +540,8 @@ def merge(a, b, path=None, conflict_resolver=None):
             a[key] = b[key]
     return a
 
+def shellquote(s):
+    return "'" + s.replace("'", "'\\''") + "'"
```

Comment:
In shell, inside single quoted strings no escapes are allowed, instead one close the string, have a single escaped `'` character and start the string again. There are other ways to do it but this one is simple and safe. For instance `foo'bar` is escaped as `'foo'\''bar'`, in this case `foo\'bar` would have been enough. Once again simplicity is preferred here. Also one more thing I started to understand shell quoting/strings once I stopped trying to map to strings in other languages...